templates/kotlin/src/main/kotlin/io/appwrite/models/Payload.kt.twig

```diff
@@ -0,0 +1,67 @@
+package io.appwrite.models
+
+import io.appwrite.extensions.gson
+import java.io.File
+import java.net.URLConnection
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class Payload private constructor() {
+
+    lateinit var path: String
+    lateinit var filename: String
+    lateinit var mimeType: String
+    lateinit var sourceType: String
+    lateinit var data: Any
```

Comment:
I see some complexity here. Is all of this needed?

For example path - do we even need to remember this? Do we send it to API?
Or sourceType - can't `data` just always be ByteArray, and in fromX methods we always convert to ByteArray? API doesn't care what type it is
mimeType - I think API doesn't need it. it only needs file name

Reply:
The path is used to access the [file](https://github.com/appwrite/sdk-generator/blob/4d2f3eaa9fcad9cc73478276ace2cd98392bbf13/templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig#L332)
[Source type](https://github.com/appwrite/sdk-generator/blob/4d2f3eaa9fcad9cc73478276ace2cd98392bbf13/templates/kotlin/src/main/kotlin/io/appwrite/Client.kt.twig#L342) is used to fetch the file before push and the code is in the client, same as for mimeType